server/src/db/filesystem_db.rs

```diff
@@ -27,6 +27,37 @@ impl FileSystemDb {
         }
     }
 
+    fn provided_entity<T: DbEntity>(&self, user_id: &Id<User>, provided_id: &Id<T>) -> bool {
```

Comment:
`provided_entity` is a noun (with an adjective). This may be a good name for a variable. But function names should almost always be verbs.

This function in particular returns `bool`. Such functions typically have names that seem to answer a yes-or-no question. `has_thing`, `is_something`, `contains_element` etc.